### ISSUE TITLE
On-board data storage solution

### ISSUE THREAD

**Issue by christiankuttke** [openUC2/openUC2_SOOP#4]
- HDD (spinning)
  - 16 € per TB at 14,18,20 TB ([Toshiba Enterprise Capacity MG07ACA](https://geizhals.de/toshiba-enterprise-mg07aca-14tb-mg07aca14te-a1812522.html?hloc=at&hloc=de))
  - 8 W power consumption at full load - produces lots of heat!
  - produces vibration
  - Power supply (12V, 5V) via small sata power adapter from 12V possible
  - big (26x147x101 mm)
- SSD
  - 20 € per TB at 2 TB https://geizhals.de/mushkin-tempest-2tb-mknssdts2tb-d8-a2631106.html?hloc=at&hloc=de
  - 3 W power consumption 
  - power supply 3.3 V is often done on adapter board
  - small (22x80x4 mm)
  - no vibrations
  - more shock resistance
  - Faster write speeds

Make a decision by specifying:

- How much data needs to be stored
- What environment will it be placed in
- How much space and ventilation will our enclosure have

**Comment by beniroquai** [comment]
We had several schemes for data storing. E.g. a RAW image every minute, JPEGs every second or even capture H.264-encoded videos. I can imagine that there is a lot of static background that could be removed. 

Let's say we produce 100MB/minute and we let the microscope run 24h*10days*60minutes*100MB. This would be ~2TB. 

Vibration is a thing. I would prefer SSDs.

**Comment by christiankuttke** [comment]
Raspberry Pi 4 has PCIe x1 lane compatible with nvme SSDs: https://www.raspberrypi.com/products/m2-hat-plus/

**Comment by beniroquai** [comment]
That's for Raspberry Pi 5 only, right?
I'm tending to go towards that one. The 4 is freezing very often. Might be
a specific thing with the model I have.
Also, the raspberry pi can have the 7 inch screen which the jetson can't. I
ordered the adapter cable from CSI to the pi5. Let's see.
Now we need power 12V => 5V or whatever works best so that there is no
voltage drop.

**Comment by christiankuttke** [comment]
Design-locking this to a SSD suitable for Raspberry Pi 5: A NVMe drive in M.2 M key, in 2230 or 2242 form factor (aka length). 

Additional hardware needed is a board which accepts the SSD and adapts the M.2 to ribbon cable to go into the Raspberry Pi 5 PCIe FFC connector.

Example: Raspberry Pi 5 M.2 HAT https://core-electronics.com.au/raspberry-pi-m2-hat.html

**Comment by beniroquai** [comment]
Question: Should we really go for a PCI solution? My fear is that, once we have to pull down the data from the device we have to rely on network. I would prefer an USB3 SSD, what do you think @ranranking @christiankuttke ?

**Comment by christiankuttke** [comment]
When there is the built-in nvme M.2 SSD, the images can be recorded quickly. This leaves all USB and the Ethernet port free. 

USB and Ethernet can be routed to face the user when the box is open. This way, users can connect a USB hard drive and a mouse to start transfer. Or, they can use the ethernet to plug it in to their network (which I don't think will happen because of university networking rules) and then use mouse/keyboard via usb to start the transfer.

For putting USB and ethernet to the inside surface of the box from wherever the Raspberry Pi is mounted, short passive extension cables or even a USB hub can be used.

**Comment by beniroquai** [comment]
Agree. good points you brought up! 
So, then let's think about something like this?
https://www.amazon.de/Crucial-Intern-5-000MB-Kompatibel-Desktop/dp/B0BYW8FLKN/ref=sr_1_3?dib=eyJ2IjoiMSJ9.PaH6Xr2mJnw7YuZabhzhl0pYI_58e732WZoygFfy9vTugVlGSh1lDwstUWm7jqOejpa1-H9eW7_mJOPOk-VuMHg9sdvCcXKvn1w1t04vTe9BXu8hMQSquxwIKJ64vmRBf5jMDXP3EzvqMavI_4Rlq-g_DTUFvBbV6Z8HN9LHWYKOJkZzEa4BpQiG-fHI5K8va-GWktFPaQCgzBcZHpSa7nLq6eSGVljnkxWXOfKpkAc.A7pitJx_V7qNNzre49MtVtDZlH_xi-EMG_ca94qgil0&dib_tag=se&keywords=m2+ssd+2tb&qid=1721595111&sr=8-3

in combination with this
https://www.amazon.de/PCIe-M-2-Raspberry-Pi-M-Key-Multicolor/dp/B0CRDY6Q3C

and then we would need to have a ethernet port for the case like this
https://www.rocketronics.de/shop/de/neu-rj45-kabeldurchfuehrung.html

**Comment by beniroquai** [comment]
I personally wouldn't stress the water-proofiness so much. AS long as some smaller droplets won't make it into the box I think we're good!

**Comment by beniroquai** [comment]
In the End we had to take an SD-Card adapter and a large SD card. @christiankuttke is this any good? Are there solutions where micro sd cards with 2TB become reliable? 
https://www.amazon.de/2TB-Micro-SD-Karte-Computer-Zubeh%C3%B6r/s?k=2TB+Micro+SD-Karte&rh=n%3A340843031

**Comment by christiankuttke** [comment]
Unfortunately no. Next best thing vs. SD cards are emmc chips. SATA or M.2 SSDs are the best. Enterprise SSDs have supercaps inside to be able to write cached sectors to non-volatile memory at unexpected power loss, but those are probably outside our price range.

**Comment by christiankuttke** [comment]
Also, cheap generic SD-card adapters are the worst. The one we had yesterday had an unreliable contact, and if that prevents the Raspi from booting or causes data loss in high-vibration environments, then that's a design fault in our product.

**Comment by beniroquai** [comment]
Agree. So we have to chase the power lack for the pi or find a way to power it externally somehow. My suspicion is that the official raspi power supply will bypass the current limitation so that more current can be drawn from the USB powers. An alternative could be an externally powered USB hub for the external devices. Are there any of these devices as "Gehäusedurchgangslösung" or IP6x standard?
